tests/e2e/pvc_utils.go

```diff
@@ -128,3 +136,273 @@ func deletePersistentVolumeClaim(
+	svpvcName string, size string, storageclass *storagev1.StorageClass,
+	pvReclaimPolicy v1.PersistentVolumeReclaimPolicy) (*v1.PersistentVolumeClaim, *v1.PersistentVolume) {
+	ginkgo.By("Creating PV in guest cluster")
+	gcPV := getPersistentVolumeSpecWithStorageclass(svpvcName, pvReclaimPolicy,
+		storageclass.Name, nil, size)
+	gcPV, err := client.CoreV1().PersistentVolumes().Create(ctx, gcPV, metav1.CreateOptions{})
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	gcPVName := gcPV.GetName()
+	framework.Logf("PV name in GC : %s", gcPVName)
+	err = fpv.WaitForPersistentVolumePhase(ctx, "Available", client, gcPVName, poll, pollTimeout)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+	ginkgo.By("Creating PVC in guest cluster")
+	gcPVC := getPVCSpecWithPVandStorageClass(svpvcName, namespace, nil, gcPVName, storageclass.Name, size)
+	gcPVC, err = client.CoreV1().PersistentVolumeClaims(namespace).Create(ctx, gcPVC, metav1.CreateOptions{})
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+	ginkgo.By("Waiting for claim to be in bound phase")
+	err = fpv.WaitForPersistentVolumeClaimPhase(ctx, v1.ClaimBound, client,
+		namespace, gcPVC.Name, framework.Poll, framework.ClaimProvisionTimeout)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+	framework.Logf("PVC name in GC : %s", gcPVC.GetName())
+
+	return gcPVC, gcPV
+}
+
+// createStaticPVandPVCandPODinGuestCluster creates static PV and PVC in guest cluster.
+func createStaticPVandPVCandPODinGuestCluster(client clientset.Interface,
+	ctx context.Context, namespace string, svpvcName string, size string, storageclass *storagev1.StorageClass,
+	pvReclaimPolicy v1.PersistentVolumeReclaimPolicy) (*v1.PersistentVolumeClaim, *v1.PersistentVolume,
+	*v1.Pod, string) {
+
+	gcPVC, gcPV := createStaticPVandPVCinGuestCluster(client, ctx, namespace,
+		svpvcName, size, storageclass, pvReclaimPolicy)
+	// Create a Pod to use this PVC, and verify volume has been attached.
+	ginkgo.By("Creating pod to attach PV to the node")
+	pod, err := createPod(ctx, client, namespace, nil, []*v1.PersistentVolumeClaim{gcPVC}, false, execCommand)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+	ginkgo.By(fmt.Sprintf("Verify volume: %s is attached to the node: %s",
+		gcPV.Spec.CSI.VolumeHandle, pod.Spec.NodeName))
+	vmUUID, err := getVMUUIDFromNodeName(pod.Spec.NodeName)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	volHandle := getVolumeIDFromSupervisorCluster(gcPV.Spec.CSI.VolumeHandle)
+	isDiskAttached, err := e2eVSphere.isVolumeAttachedToVM(client, volHandle, vmUUID)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	gomega.Expect(isDiskAttached).To(gomega.BeTrue(), "Volume is not attached to the node")
+
+	return gcPVC, gcPV, pod, vmUUID
+
+}
+
+// onlineVolumeResizeCheck this method increases the PVC size, which is attached
+// to POD.
+func onlineVolumeResizeCheck(f *framework.Framework, client clientset.Interface,
```

Comment:
Can you please move this util to a seperate util exclusively for volume expansion